src/bandersnatch/tests/plugins/test_allowlist_name.py

```diff
@@ -107,6 +107,28 @@ def test__filter__nomatch_package(self) -> None:
         self.assertIn("foo", mirror.packages_to_sync.keys())
         self.assertNotIn("foo2", mirror.packages_to_sync.keys())
 
+    def test__filter__name_only(self) -> None:
+        mock_config(
+            """\
+[mirror]
+storage-backend = filesystem
+
+[plugins]
+enabled =
+    allowlist_project
+
+[allowlist]
+packages =
+    foo==1.2.3
+"""
+        )
+
+        mirror = Mirror(Path("."), Master(url="https://foo.bar.com"))
+        mirror.packages_to_sync = {"foo": "4.5.6"}
+        mirror._filter_packages()
+
+        self.assertIn("foo", mirror.packages_to_sync.keys())
```

Comment:
Am I misunderstanding, The allowlist only allows version 1.2.3 shouldn’t foo not exist in the keys() generator/list ?

Reply:
This is to test the bug I believe currently exists (and my fix for it). `AllowlistProject` sates that it only cares about package names and not versions ("This plugin only processes packages, if the line in the packages configuration contains a PEP440 specifier it will be processed by the allowlist release filter").  Currently though it stores the whole filter line including any version specifier, and so will filter any packages not matching that exact name+version.

The corresponding code to this test is [here](https://github.com/pypa/bandersnatch/pull/632/files#diff-9b88701e7c64d1876776b5fb0b0624d5L61-R63)

Reply:
I'm maybe confusing myself over what the intended behaviour is here, but with only the `allowlist_project` plugin enabled, only the package names should be checked? Enabling the `allowlist_release` plugin will then also check the versions? The docs don't specifically mention what the difference is.

Reply:
Yes @rkm, you are correct, allowlist_project filters only on package names and allowlist_release enables filtering for versions.  The filtering occurs in two stages, before the metadata is downloaded and after the metadata is downloaded.  Project filters (allowlist_project and blocklist_project) occur before the metadata of the package's are downloaded and thus reduce the amount of data that needs to be synced, but they can only filter off of the package's name since you need the metadata to filter off of release info like the version specifiers.  Technically allowlist_release and blocklist_release can perform the same job of their project variants, but you lose some efficiency if you don't care about the versions.